netket/optimizer/sr/api.py

```diff
@@ -175,8 +178,14 @@ def _SR(
     if qgt is None:
         qgt = QGTAuto(solver)
 
+    if isinstance(diag_shift, Number):
+        diag_shift = constant_schedule(diag_shift)
+
     return SR(
-        partial(qgt, diag_shift=diag_shift, **kwargs),
+        lambda vstate, step_value: qgt(
+            vstate, diag_shift=diag_shift(step_value), **kwargs
+        ),
```

Comment:
@inailuig recently added support for a `diag_shift` that is a pytree, that a student is using and I wouldn't want to lose this support as it might be useful in the future.

I think we can rather check that if `diag_shift` is not an optax type (If they signal such thing with a type) we don't wrap it, otherwise we wrap it in `constant_schedule`.

Reply:
How about this: if it's not a `Callable`, we wrap it with a generalisation of `constant_schedule` that handles anything (IIRC, `constant_schedule` simply returns `lambda _: input` so that should be good enough already). That way, we can treat @inailuig's addition and the basic `diag_shift` on the same footing (and even allow a step-dependent pytree `diag_shift`)

Reply:
> @inailuig recently added support for a `diag_shift` that is a pytree, that a student is using and I wouldn't want to lose this support as it might be useful in the future.

BTW, is this available somewhere?

Reply:
In the last tagged release, a few days ago.
Just try running `vstate.quantum_Geometric_tensor(nk.optimizer.qgt.QGTOnTheFly(diag_shift=vstate.parameters))` for example, and it should work.